### PR TITLE
Fix/mkt 2696 clone adding custom limit parameter for gatsby build

### PR DESCRIPTION
**Pull request by Contentstack-AnkitaD**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=k_r3l3q)

- Added the support to make limit value {No. of entries to be fetched at a time} configurable.

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -193,11 +192,13 @@ const getPagedData = async (
   responseKey,
   query = {},
   skip = 0,
-  limit = 100,
+  limit = config?.limit || 50,
```
Comment (abhishek305) [inline src/fetch.js:195]: you dont need to provide 50 here  as its already configured plugin option schema file as default value please verify and add changes

File: README.md [hunk 2]
```diff
@@ -71,22 +68,24 @@ plugins: [
...
       // Optional: Specify true to convert the JSON-RTE repsonse to HTML. Default it is set to false
       jsonRteToHtml: false ,
-      
+
       // Optional: Specify a number to perform http-retries for network failures. By default it is set to 3.
-      httpRetries: 2
+      httpRetries: 2,
+
+      //Specify the number of entries/assets to be fetched per page. Maximum limit accepted is 100.
+      limit: 50
```
Comment (abhishek305) [inline README.md:76]: also mention that default is kept to 50

**Review by abhishek305 — CHANGES_REQUESTED** [review]
Few points to check
- bump package version(change needed)
- validate there are no missing data with respect to the stack

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gatsby-source-contentstack",
-  "version": "5.1.0",
+  "version": "5.2.0",
```
Comment (abhishek305) [inline package.json:3]: make it to 5.1.1

**Review by abhishek305 — CHANGES_REQUESTED** [review]

**Review by surajcontentstack — APPROVED** [review]

**Comment by Contentstack-AnkitaD** [comment]
> do we have unit testes for this repo? @Contentstack-AnkitaD

No.

**Comment by Contentstack-AnkitaD** [comment]
Closing this PR as I need more time for testing... Will merge it once testing is completed

**Pull request merged**